CHANGELOG.md

```diff
@@ -1,3 +1,259 @@
+
+2. Export your platform name and assets directory name used in your platform configuration. It will be used in next steps:
+  ```sh
+  export PLATFORM="packet" && export ASSETS_DIR="assets"
+  ``` 
+
+3. Remove old asset files:
+  ```sh
+  rm $ASSETS_DIR/terraform-modules/$PLATFORM/flatcar-linux/kubernetes/require.tf \
+  $ASSETS_DIR/terraform-modules/$PLATFORM/flatcar-linux/kubernetes/workers/require.tf \
+  $ASSETS_DIR/terraform-modules/dns/route53/require.tf 2> /dev/null
+  ```
+
+4. Go to terraform directory:
+  ```sh
+  cd $ASSETS_DIR/terraform
+  ```
+
+5. Replace old providers:
+  ```sh
+  terraform state replace-provider -auto-approve registry.terraform.io/-/ct registry.terraform.io/poseidon/ct && \
+  terraform state replace-provider -auto-approve registry.terraform.io/-/template registry.terraform.io/hashicorp/template
+  ```
+
+6. **NOTE:** delete old ReplicaSet `kube-apiserver` only for single node cluster.
+  ```sh
+  kubectl delete rs $(kubectl get rs -n kube-system -l k8s-app=kube-apiserver --no-headers=true  | sort -k5 -h | head -n +2 | awk '{print $1}')
+  ```
+
+7. FelixConfiguration to be managed by Helm:
+  ```sh
+  kubectl label FelixConfiguration default app.kubernetes.io/managed-by=Helm && \
+  kubectl annotate FelixConfiguration default meta.helm.sh/release-name=calico && \
+  kubectl annotate FelixConfiguration default meta.helm.sh/release-namespace=kube-system
+  ```
+
+Finally, run:
+
+```sh
+lokoctl cluster apply --skip-components -v
+```
+
+The update process typically takes about 10 minutes.
+After the update, running `lokoctl health` should result in an output similar to the following:
+
+```sh
+Node                     Ready    Reason          Message
+
+lokomotive-controller-0  True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-0    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-1    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-2    True     KubeletReady    kubelet is posting ready status
+Name      Status    Message              Error
+
+etcd-0    True      {"health":"true"}
+```
+
+#### Cluster nodes component upgrade (optional)
+
+- Manually upgrade etcd following the steps mentioned in the doc
+  [here](https://github.com/kinvolk/lokomotive/blob/v0.5.0/docs/how-to-guides/upgrade-etcd.md).
+- Manually upgrade the kubelet running on the nodes, by following the steps mentioned in the doc
+  [here](https://github.com/kinvolk/lokomotive/blob/v0.5.0/docs/how-to-guides/upgrade-bootstrap-kubelet.md).
+
+#### cert-manager Upgrade steps
+
+These steps are curated from the upgrade doc provided by [cert-manager](https://cert-manager.io/docs/installation/upgrading/upgrading-0.16-1.0/#kubernetes-1-16-and-above):
+
+Apply manifest:
+
+  ```sh
+  kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v1.0.3/cert-manager.yaml
+  ```
+
+Now it is safe to update:
```

Comment:
This implies that without the steps above it's *not* safe to update. I'd include a short sentence in the beginning of this section explaining the reasoning behind it. Example:

"cert-manager v1.2.3 introduces a backward-incompatible feature which requires doing so and so manually to avoid this and that."